impeller/entity/geometry/stroke_path_geometry.h

```diff
@@ -9,6 +9,11 @@
 
 namespace impeller {
 
+class VertexWriter {
+ public:
+  virtual void AppendVertex(const Point& point) = 0;
```

Comment:
This is introducing runtime polymorphism as a way to share logic, but it looks like runtime polymorphism isn't required.  This runtime polymorphism is also O(n) against the number of points.

Alternatively, just remove this superclass and switch `StrokePathGeometry` to be templated across which type it is using.

Reply:
Can we do that without moving all of this logic to the header?

Reply:
I looked at this a bit, pretty much everything would need to become templated. I'm not incredibly against it, but the virtual method call isn't close to the most expensive thing here so i'd like to punt on it for now.

Reply:
> Can we do that without moving all of this logic to the header?

Sometimes you can just put everything in a .cc file in an anonymous namespace.  Not sure if you could do that here.

> I looked at this a bit, pretty much everything would need to become templated. I'm not incredibly against it, but the virtual method call isn't close to the most expensive thing here so i'd like to punt on it for now.

If it's difficult to do that sounds good to me.